run/src/main/kotlin/com/cosmotech/run/service/RunServiceImpl.kt

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.run.service
+
+import com.cosmotech.api.CsmPhoenixService
+import com.cosmotech.api.events.RunStart
+import com.cosmotech.api.events.RunStop
+import com.cosmotech.api.rbac.CsmRbac
+import com.cosmotech.api.rbac.PERMISSION_DELETE
+import com.cosmotech.api.rbac.PERMISSION_READ
+import com.cosmotech.api.rbac.PERMISSION_WRITE
+import com.cosmotech.api.rbac.getScenarioRolesDefinition
+import com.cosmotech.api.utils.constructPageRequest
+import com.cosmotech.run.RunApiServiceInterface
+import com.cosmotech.run.RunContainerFactory
+import com.cosmotech.run.container.StartInfo
+import com.cosmotech.run.domain.Run
+import com.cosmotech.run.domain.RunLogs
+import com.cosmotech.run.domain.RunState
+import com.cosmotech.run.domain.RunStatus
+import com.cosmotech.run.domain.RunTemplateParameterValue
+import com.cosmotech.run.repository.RunRepository
+import com.cosmotech.run.utils.isTerminal
+import com.cosmotech.run.utils.withoutSensitiveData
+import com.cosmotech.run.workflow.WorkflowService
+import com.cosmotech.runner.RunnerApiServiceInterface
+import com.cosmotech.runner.domain.Runner
+import com.cosmotech.runner.service.getRbac
+import java.time.Instant
+import org.springframework.context.event.EventListener
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+
+internal const val WORKFLOW_TYPE_RUN = "container-run"
+
+@Service
+@Suppress("TooManyFunctions")
+class RunServiceImpl(
+    private val containerFactory: RunContainerFactory,
+    private val workflowService: WorkflowService,
+    private val runnerApiService: RunnerApiServiceInterface,
+    private val runRepository: RunRepository,
+    private val csmRbac: CsmRbac
+) : CsmPhoenixService(), RunApiServiceInterface {
+
+  override fun listRuns(
+      organizationId: String,
+      workspaceId: String,
+      runnerId: String,
+      page: Int?,
+      size: Int?
+  ): List<Run> {
+    // This call verify by itself that we have the read authorization in the scenario
+    val runner = runnerApiService.getRunner(organizationId, workspaceId, runnerId)
+    csmRbac.verify(runner.getRbac(), PERMISSION_READ, getScenarioRolesDefinition())
```

Comment:
Can it be done in `getRunner` function ?

Reply:
it's already done in the getRunner function.
check removed